### PR TITLE
fix(review): render full @UserMessage templates so AI prompts carry every context variable

### LINKED ISSUE #186: fix(review): AI prompts drop every context variable but the first (@UserMessage on a parameter)
### Problem Statement

Every `@RegisterAiService` interface declares `@UserMessage(...)` on its **first method parameter** instead of on the method:

```java
@SystemMessage(PrReviewPrompts.SYSTEM)
TokenStream reviewStream(
    @UserMessage(PrReviewPrompts.USER) @V("diff") String diff,   // ← on the parameter
    @V("prContext") String prContext,
    @V("repoInstructions") String repoInstructions,
    ...);
```

quarkus-langchain4j treats a **parameter-level** `@UserMessage` as *"this parameter is the user message"* — it uses that parameter's raw value as the entire user message and **never renders the template**. As a result only the first `@V` ever reaches the model; every other variable is built, escaped… and silently discarded.

### Impact

| Service | Reaches the model | Silently dropped |
|---|---|---|
| `PrReviewer` | the diff | repository instructions (`.github/thrillhousebot.md`), project stack, PR title/description, base comparison, related tests, **previous findings** |
| `FindingVerifier` | the candidate findings | **the diff**, project stack, previous findings |
| `ReplyAssistant` | the maintainer's question | the diff, the finding under discussion, the thread, PR context |

Consequences: reviews ignore the repo's own review guidance and re-raise previously-answered findings (the in-prompt `previous_findings` was never sent — only the Java-side backstop caught it); the second-pass verifier audits candidates **without the diff**; and `@thrillhousebot` replies answer "I can't see the diff" to legitimate questions about the PR.

### Evidence

Captured from the running bot with `quarkus.langchain4j.openai.chat-model.log-requests=true`:

- A `/review` request's `user` message is **only** the formatted diff (`## Overview: 6 files (+385 -12)…`) — no `## Repository instructions` or previous-findings sections, even though the same run logged `Using instructions file: .github/thrillhousebot.md (3086 bytes)` and `Project stack … 712 chars` moments earlier.
- A conversational `@thrillhousebot does it introduce any breaking changes?` request's `user` message is the bare question (317 prompt tokens, system + one line); the model replies that it cannot see the diff or PR description.

### Root Cause

`@UserMessage` on a parameter ⇒ parameter value used verbatim as the user message ⇒ the multi-variable template is never rendered ⇒ all other `@V` values are dropped. The reply model "looked" guarded against prompt injection only because the off-topic refusals (cake recipe, code-gen) don't need the diff.

### Affected Versions

Pre-existing — **not** introduced in v0.2.0:
- `PrReviewer` and `FindingVerifier` have had the parameter-level `@UserMessage` since the initial **v0.1.0** release (commit `518c905`).
- `ReplyAssistant` since the conversational-replies feature (#31, v0.2.0).

### Fix

PR #184:
- Move `@UserMessage` to the **method** on all three services so every `@V` is interpolated.
- Reduce `PromptTemplateEscaper.escape()` to marker-neutralization only: with method-level rendering `@V` values are bound as Qute **data** (not re-parsed), so they reach the model byte-exact and uninterpreted; the old `{|…|}` unparsed-section wrapper relied on a second Qute pass that never happens and was corrupting content containing `|}`.
- Regression coverage: `AiServicePromptRenderingTest` (drives the real engine via a recording `ChatModel` and asserts each prompt carries every variable, with byte-exact / injection-marker edge cases) and `AiServiceUserMessagePlacementTest` (structural guard pinning `@UserMessage` to the method).

> [!NOTE]
> The open #176 (`/describe`) PR's `PrDescribeAssistant` has the same parameter-level pattern and needs the same fix on that branch.

### Priority

Critical — silently degraded every review and broke conversational replies.

### Code of Conduct

- [x] I have searched for existing issues and this is not a duplicate.

### PR DESCRIPTION
**Pull request by devops-thiago**
Closes #186

## What type of PR is this?

- [x] 🐛 Bug Fix
- [x] ✅ Test

## The bug (critical)

Every AI service declared `@UserMessage` on the **first method parameter**:

```java
String reply(
    @UserMessage(ReplyAssistantPrompts.USER) @V("question") String question, // ← here
    @V("prContext") String prContext, ...);
```

quarkus-langchain4j treats a **parameter-level** `@UserMessage` as "*this parameter is the user message*" — it uses that parameter's raw value and **never renders the template**. So only the first `@V` ever reached the model; every other variable was silently dropped:

| Service | Reached the model | Silently dropped |
|---|---|---|
| `PrReviewer` | the diff | repo instructions (`.github/thrillhousebot.md`), project stack, PR title/description, base comparison, related tests, **previous findings** |
| `FindingVerifier` | the findings | **the diff**, project stack, previous findings |
| `ReplyAssistant` | the maintainer's question | the diff, the finding, the thread, PR context |

Reviews still produced output (the diff is the bulk of the signal), which is why this went unnoticed — but they ran blind to the repo's own review guidance and prior findings, the verifier audited without the diff, and `@thrillhousebot` questions got "I can't see the diff" answers.

### Evidence

Captured from the running bot with `log-requests=true` — the `/review` user message is **only** the diff (`## Overview: 6 files…`), with no `## Repository instructions` / previous-findings sections, even though the logs show both were fetched moments earlier.

## The fix

- Move `@UserMessage` to the **method** on `PrReviewer`, `ReplyAssistant`, `FindingVerifier`, so every `@V` is interpolated.
- Reduce `PromptTemplateEscaper.escape()` to **marker-neutralization only**. With method-level rendering, `@V` values are bound as Qute **data** (not re-parsed), so they reach the model byte-exact and uninterpreted. The old `{|…|}` unparsed-section wrapper relied on a second Qute pass that does not happen for data-bound values — it leaked into the prompt literally and corrupted any content containing `|}`. The quote validator and `DiffLineResolver` already compare against `neutralizeMarkers(diff)`, so they stay aligned.

All of this was verified against the **real engine** (not assumed) via a recording `ChatModel` probe before settling the approach.

## Tests

- **`AiServicePromptRenderingTest`** — drives the real quarkus-langchain4j pipeline with a recording model and asserts each rendered prompt carries **every** `@V` variable; plus edge cases: blank optional sections are omitted, hostile content (Qute syntax, `|}`, backslashes) renders byte-exact and uninterpreted, and spoofed `<<<DIFF_END>>>` markers are neutralized. Each test fails against the old code.
- **`AiServiceUserMessagePlacementTest`** — fast structural guard pinning `@UserMessage` to the method (not a parameter) on every service.
- Updated `PromptTemplateEscaperTest` / `FindingVerificationServiceTest` to the new escaping contract; removed the obsolete `PromptTemplateEscaperQuteTest`.

Full suite: **1180 tests, 0 failures**; spotless + spotbugs clean.

## Notes

- `PrDescribeAssistant` (the still-open #176 `/describe` PR) has the **same** parameter-level `@UserMessage` pattern and will need the same one-line fix there.

### CODE REVIEW

File: src/test/java/dev/thiagogonzaga/thrillhousebot/review/ai/AiServicePromptRenderingTest.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            () ->
+                replyAssistant.reply(
+                    PromptTemplateEscaper.escape("Q"),
+                    "",
+                    "",
+                    PromptTemplateEscaper.escape(hostile),
+                    ""));
+
+    assertTrue(user.contains("{config:secret}"), "Qute expression must not be interpreted");
+    assertTrue(user.contains("{#if x}IF{/if}"), "Qute section must not be interpreted");
```
[the remark this answers is not in the document]
Comment (devops-thiago) [reply src/test/java/dev/thiagogonzaga/thrillhousebot/review/ai/AiServicePromptRenderingTest.java:122]: False positive — no change. The two assertions check different strings: `<<DIFF_END>> after` (2 angles, present after neutralization) and `<<<DIFF_END>>> after` (3 angles, absent). They look identical only because this repo's marker-neutralization rewrites the literal 3-angle marker in the test source down to 2 angles in the diff under review — the self-referential edge documented in PromptTemplateEscaper. The test passes (full suite green; SonarCloud 100% coverage on new code). Tracked as #187.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/test/java/dev/thiagogonzaga/thrillhousebot/review/ai/AiServicePromptRenderingTest.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        captureBlocking(
+            () ->
+                replyAssistant.reply(
+                    PromptTemplateEscaper.escape("Q"),
+                    "",
+                    "",
+                    PromptTemplateEscaper.escape(hostile),
+                    ""));
+
+    assertTrue(user.contains("{config:secret}"), "Qute expression must not be interpreted");
```
[the remark this answers is not in the document]
Comment (devops-thiago) [reply src/test/java/dev/thiagogonzaga/thrillhousebot/review/ai/AiServicePromptRenderingTest.java:121]: False positive — no change. The two assertions check different strings: `<<DIFF_END>> after` (2 angles, present after neutralization) and `<<<DIFF_END>>> after` (3 angles, absent). They look identical only because this repo's marker-neutralization rewrites the literal 3-angle marker in the test source down to 2 angles in the diff under review — the self-referential edge documented in PromptTemplateEscaper. The test passes (full suite green; SonarCloud 100% coverage on new code). Tracked as #187.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/test/java/dev/thiagogonzaga/thrillhousebot/review/ai/AiServicePromptRenderingTest.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    assertTrue(user.contains("{#if x}IF{/if}"), "Qute section must not be interpreted");
+    assertTrue(user.contains("a|}b"), "section terminator must not be doubled");
+    assertTrue(user.contains("backslash\\n"), "backslash must survive verbatim");
+    assertTrue(user.contains("<<DIFF_END>> after"), "spoofed end marker must be neutralized");
+    assertFalse(user.contains("<<<DIFF_END>>> after"), "spoofed end marker left un-neutralized");
```
[the remark this answers is not in the document]
Comment (devops-thiago) [reply src/test/java/dev/thiagogonzaga/thrillhousebot/review/ai/AiServicePromptRenderingTest.java:126]: False positive — no change. The two assertions check different strings: `<<DIFF_END>> after` (2 angles, present after neutralization) and `<<<DIFF_END>>> after` (3 angles, absent). They look identical only because this repo's marker-neutralization rewrites the literal 3-angle marker in the test source down to 2 angles in the diff under review — the self-referential edge documented in PromptTemplateEscaper. The test passes (full suite green; SonarCloud 100% coverage on new code). Tracked as #187.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/test/java/dev/thiagogonzaga/thrillhousebot/review/ai/FindingVerificationServiceTest.java [hunk 4]
```diff
@@ -415,9 +416,13 @@ void shouldSendEscapedCandidatesWithIdsAndPassThroughContext() {
     var candidates = ArgumentCaptor.forClass(String.class);
```
[the remark this answers is not in the document]
Comment (devops-thiago) [reply src/test/java/dev/thiagogonzaga/thrillhousebot/review/ai/FindingVerificationServiceTest.java:416]: False positive — no change. `assertTrue` checks `<<DIFF_END>> tail` (2 angles, present after neutralization) and `assertFalse` checks `<<<DIFF_END>>>` (3 angles, absent) — distinct strings. They look identical only because marker-neutralization rewrites the literal 3-angle marker in the test source to 2 angles in the diff under review (the self-referential edge documented in PromptTemplateEscaper). The test passes. Tracked as #187.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
